### PR TITLE
GCN for solving Facebook large page-page Network multi-class classification problem.

### PR DESCRIPTION
**Pull request by Sin4ik02**
Created by student: 47421347
This is a multi-layer Graph Convolutional Network PyTorch solution for the Facebook large page-page Network multi-class classification problem. Achieved test accuracy is **92.65%**

The directory has the following structure

- `modules.py` - where the model architecture is declared
- `dataset.py` - where the dataset is loaded and preprocessed
- `train.py` - where the model is trained and saved
- `utils.py` - stores the helper functions used for plotting the results
- `predict.py` - loads the previously saved model and tests it
- `images/` - directory, containing the plots
- `README.md` - contains the description of the problem and the model.

### CODE REVIEW

**Comment by gayanku** [comment]
**This is an initial inspection, no action is required at this point**
GNN FB Page-Page Network dataset ---- Normal Difficulty
| Category	| | Marks | Comments |	
| :---  | :---: | :---: | :---  |		
| Algorithm solves the problem 		| 5 | 5 | |
| Implementation functions as intended	| 3 | 3 | 3 layer GCN, Own GNNLayer impl. |
| Good design 					| 1 | 1 | Modular, Reusable |
| Commenting 					| 1 | 1 | Meaningful docstrings and comments |
| Algorithm above Normal Difficulty 	| 5 | 5 | |
| Algorithm is Hard difficulty 			| 5 | 0 | Normal Difficulty |
| Section IV : Max mark 15 from 20		|   | **15**   | |

**TSNE / UMAP:** 
Good.
- Discussion: Good, with a bit more depth useful.

**Suggestions:**
Setup/Reproducibility/Requirements would be good to have.
Could add a hyper parameter search or discuss how the values were obtained.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. Minimalistic.</td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>Good solution to problem. </td><td></td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Normal. GNN Task.</td><td>-5</td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Good. </td><td></td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. Give a  high level intro to the approach instead of the file list.</td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-7</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by hanemma7moud** [comment]
Ivan Sinitsyn

**Comment by Sin4ik02** [comment]
> Ivan Sinitsyn

Yeah that's me
